### PR TITLE
Add more tests for TokenAssociateTransaction

### LINKED ISSUE #1443: More Unit tests for `TokenAssociateTransaction`
### Problem

`TokenAssociateTransaction` is missing some Unit tests.

### Solution

To add the following Unit tests in `TokenAssociateTransactionTest.java`:

- `ConstructTokenAssociateTransactionFromTransactionBodyProtobuf`
- `GetSetAccountId`
- `GetSetAccountIdFrozen`
- `GetSetTokenIds`
- `GetSetTokenIdFrozen`

For reference see `TokenAssociateTransactionTest.cc` from the [C++ SDK](https://github.com/hashgraph/hedera-sdk-cpp).

### Alternatives

N/A

### PR DESCRIPTION
**Pull request by dikel**
**Description**:
Adds more tests for `TokenAssociateTransaction`

**Related issue(s)**:

Fixes #1443 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java [hunk 1]
```diff
@@ -84,4 +86,57 @@ void fromScheduledTransaction() {
...
+            TokenId.fromString("10.11.12"));
+
+        var transactionBody = TokenAssociateTransactionBody.newBuilder()
+            .addAllTokens(tokenIds.stream().map(TokenId::toProtobuf).toList())
+            .setAccount(accountId.toProtobuf()).build();
+        var txBody = TransactionBody.newBuilder().setTokenAssociate(transactionBody).build();
+        var tokenAssociateTransaction = new TokenAssociateTransaction(txBody);
+
+        assertThat(tokenAssociateTransaction.getAccountId()).isEqualTo(accountId);
+        assertThat(tokenAssociateTransaction.getTokenIds()).isEqualTo(tokenIds);
```
Comment (thenswan) [inline sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java:104]: C++ SDK tests against size of tokenIds [here](https://github.com/hashgraph/hedera-sdk-cpp/blob/63b7e0300795c2a402308841bfad1fc9ff988f46/sdk/tests/unit/TokenAssociateTransactionUnitTests.cc#L68)

I understand that this option in PR should work as well, but I mean should we make completely matching tests for the sake of consistency between SDK codebases or no?

File: sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java [hunk 2]
```diff
@@ -84,4 +86,57 @@ void fromScheduledTransaction() {
...
+    }
+
+    @Test
+    void getSetAccountId() {
+        var accountId = AccountId.fromString("1.2.3");
```
Comment (thenswan) [inline sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java:109]: Should we make some rule for unit tests that we instantiate such vars at the top of the file (like in C++ SDK or in my PRs)?
Here's couple reasons of that:
- code consistency (we're syncing w/ @dikel on couple guideline rules, so we make almost identical PRs for unit tests)
- less repeatable code

but .. we just can refactor it afterwards as well, so would be waiting here for POV of other reviewers:)

**Comment by thenswan** [comment]
Comments in my review are non-blocking and just nice-to-have/good-to-discuss stuff imo.

**Review by thenswan — COMMENTED** [review]
Comments in my review are non-blocking and just nice-to-have/good-to-discuss stuff imo.

File: sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java (hunk 1 above)
Comment (deyanzz) [reply sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java:104]: I agree, we should make completely matching tests where it's possible for the `C++` and the `Java` SDKs.
There should be added an assert to check the size of `tokenIds`.

File: sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java (hunk 2 above)
Comment (deyanzz) [reply sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java:109]: I agree with that. We should follow the `DRY (Don't Repeat Yourself)` principle where it's possible.

**Review by deyanzz — CHANGES_REQUESTED** [review]

File: sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java [hunk 3]
```diff
@@ -84,4 +86,57 @@ void fromScheduledTransaction() {
 
         assertThat(tx).isInstanceOf(TokenAssociateTransaction.class);
     }
+
+    @Test
+    void ConstructTokenDeleteTransactionFromTransactionBodyProtobuf() {
```
[the remark this answers is not in the document]
Comment (thenswan) [reply sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java:91]: I agree, good catch.

[the remark this answers is not in the document]
Comment (thenswan) [reply sdk/src/test/java/com/hedera/hashgraph/sdk/TokenAssociateTransactionTest.java:94]: @dikel PR is still missing this little part -- everything else LGTM

**Review by thenswan — COMMENTED** [review]
PR is still missing little part with `camelCase` naming -- everything else LGTM.

**Comment by thenswan** [comment]
LGTM

**Review by thenswan — APPROVED** [review]
LGTM

**Review by deyanzz — APPROVED** [review]
LGTM. All Unit tests from C++ SDK were satisfied.

**Pull request merged**